docs/user/registering-microshift-devices-acm/registering-microshift-devices-acm.md

```diff
@@ -0,0 +1,184 @@
+# Auto-Registering Devices with MicroShift into ACM
+
+If you have fleets of devices running an OS image that includes MicroShift, you can configure these fleets to auto-register MicroShift clusters with Red Hat Advanced Cluster Management (ACM).
+
+Auto-registration relies on ACM's [agent registration](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.12/html/clusters/cluster_mce_overview#importing-managed-agent) method for importing clusters. That method allows fetching the Kubernetes resource manifests to install ACM's klusterlet agent and registering the agent through calls to a REST API. This REST API can be set up as configuration source for devices by creating a Repository resource and referencing that resource from the fleet's device template.
+
+This document is intended for ACM and FlightCtl developers, QE engineers, and users who want to explore this feature. It provides step-by-step instructions, assuming minimal prior knowledge of ACM or FlightCtl.
+
+## 1 Enable `edge-manager` in ACM
+
+### 1.1 Patch `edge-manager-preview` in MCH.
+
+```console
+oc patch mch multiclusterhub -n open-cluster-management \
+       --type=json -p='[{"op": "add", "path": "/spec/overrides/components/-","value":{"name":"edge-manager-preview","enabled":true}}]'
+```
+
+[!NOTE] If you're testing with a [downstream ACM](https://quay.io/repository/acm-d/acm-custom-registry?tab=tags), you also need to add the following annotation to the MCH, otherwise agents will not be able to use downstream images.
+
+```console
+oc annotate mch multiclusterhub -n open-cluster-management mch-imageRepository='quay.io:443/acm-d'
+```
+
+### 1.2 Enable flightctl-plugin from Console.
+
+Navigate to Home -> Overview -> DynamicPlugins -> flightctl-plugin and enable it. After a few seconds, the console will prompt you to refresh the page.
+
+Once refreshed, you will see the `Edge Management` option in the left navigation menu.
+
+## 2 Setup and Configure Flightctl
+
+### 2.1 Install flightctl CLI
+
+For detailed instructions on how to install the flightctl CLI, please refer to the [Installing the Flight Control CLI](../getting-started.md#installing-the-flight-control-cli) section in the Getting Started guide.
+
+### 2.2 Login to flightctl.
+
+```console
+export FLIGHTCTL_SERVER_ADDRESS=$(oc get route flightctl-api-route -n open-cluster-management -o jsonpath='{.spec.host}')
+
+flightctl login $FLIGHTCTL_SERVER_ADDRESS --insecure-skip-tls-verify --token=<cluster admin token>
+```
+
+After flightctl is enabled, the Repository `acm-registration` will be created automatically.
```

Comment:
prefer to move this to section 1.2 since after we enable the RHEM in ACM, it should have acm-registration created, it's not related to the CLI usage part.